packages/cli-kit/src/private/node/session/identity-token-validation.ts

```diff
@@ -1,40 +1,67 @@
 import {identityFqdn} from '../../../public/node/context/fqdn.js'
 import {outputDebug} from '../../../public/node/output.js'
 import {shopifyFetch} from '../../../public/node/http.js'
+import {cacheFetch} from '../conf-store.js'
 
-export async function validateIdentityToken(token: string) {
+class NotFoundError extends Error {
```

Comment:
`Error` is not designed to be subclassed. If you try to do things like `instanceof` to check if an error is an instance of some class it'll yield unexpected values. I'd consider using [`ExtendableError`](https://github.com/Shopify/cli/blob/main/packages/cli-kit/src/public/node/error.ts#L7) or better, adjust the interface of this function to use the `Result` type, which is a better option to delegate to the caller what to do based on the obtained result.